bindings/rust/extended/s2n-tls/src/cert_authorities.rs

```diff
@@ -0,0 +1,592 @@
+        assert!(pair.server.selected_cert().is_none());
+        assert!(pair.client.selected_cert().is_none());
+
+        pair.handshake()?;
+
+        for conn in [&pair.server, &pair.client] {
+            let chain = conn.selected_cert().unwrap();
+            assert_eq!(chain.len(), 1);
+            for cert in chain.iter() {
+                let cert = cert?;
+                let cert = cert.der()?;
+                assert!(!cert.is_empty());
+            }
+        }
+
+        // Server selected the default cert
+        assert_eq!(
+            pair.server
+                .selected_cert()
+                .unwrap()
+                .iter()
+                .next()
+                .unwrap()?
+                .der()?,
+            CertKeyPair::default()
+                .into_certificate_chain()
+                .iter()
+                .next()
+                .unwrap()?
+                .der()?
+        );
+
+        // Client selected the custom cert
+        assert_eq!(
+            pair.client
+                .selected_cert()
+                .unwrap()
+                .iter()
+                .next()
+                .unwrap()?
+                .der()?,
+            keypair
+                .into_certificate_chain()
+                .iter()
+                .next()
+                .unwrap()?
+                .der()?
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn ca_list_empty() -> Result<(), Box<dyn std::error::Error>> {
+        let keypair = CertKeyPair::default();
+        let cas = Arc::new(Mutex::new(vec![]));
+        let config = {
+            let mut config = config_builder(&security::DEFAULT_TLS13)?;
+            config.set_client_auth_type(ClientAuthType::Required)?;
+            config.set_certificate_request_callback(ExtractCallback {
+                certificate: keypair.into_certificate_chain(),
+                cas: cas.clone(),
+            })?;
+            config.build()?
+        };
+
+        let mut pair = TestPair::from_config(&config);
+
+        // None before handshake...
+        assert!(pair.server.selected_cert().is_none());
+        assert!(pair.client.selected_cert().is_none());
+
+        pair.handshake()?;
+
+        let cas = cas.lock().unwrap();
+
+        // No CAs are sent by default.
+        assert_eq!(cas.len(), 0);
+
+        Ok(())
+    }
+
+    #[test]
+    fn ca_list() -> Result<(), Box<dyn std::error::Error>> {
+        let keypair = CertKeyPair::default();
+        let cas = Arc::new(Mutex::new(vec![]));
+        let config = {
+            let mut config = config_builder(&security::DEFAULT_TLS13)?;
+            config.set_client_auth_type(ClientAuthType::Required)?;
+            config.set_certificate_authorities_from_trust_store()?;
+            config.set_certificate_request_callback(ExtractCallback {
+                certificate: keypair.into_certificate_chain(),
+                cas: cas.clone(),
+            })?;
+            config.build()?
```

Comment:
System certificates [are loaded by default](https://github.com/aws/s2n-tls/blob/966deac93d934a22871ef5a724140f8b4e11dff3/bindings/rust/extended/s2n-tls/src/config.rs#L218). I'm surprised only 1 CA is loaded in this test. Is it maybe not finding system certificates on the CI runners? To make sure this works locally we should probably disable system certs.
```suggestion
            config.with_system_certs(false)?;
            config.build()?
```

Reply:
They're loaded by default, but they're not present at the time of set_certificate_authorities_from_trust_store which is what matters here. But this seems fine.